core/src/main/java/com/google/errorprone/internal/SandboxedClassLoaderRunner.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.errorprone.internal;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Executes a callback inside a special classloader that searches its own resource path _before_
+ * delegating to the runtime classpath. This allows error-prone's javac8 to be run on JDK7, and is
+ * morally equivalent to invoking java with {@code -Xbootclasspath/p:<path to javac.jar>}.
+ *
+ * @author cushon@google.com
+ */
+public class SandboxedClassLoaderRunner {
```

Comment:
nit: maybe rename to something like NonDelegatingClassLoaderRunner?  "Sandboxed" implies that this is security related.